test/generator/define_extern_opencl_generator.cpp

```diff
@@ -0,0 +1,34 @@
+#include "Halide.h"
+
+namespace {
+
+class DefineExternOpenCLOutput : public Halide::Generator<DefineExternOpenCLOutput> {
+public:
+    Input<Buffer<int32_t>> input{ "input", 1 };
+    Output<Func> output{ "output", Int(32), 1 };
+
+    Var x{"x"};
+    // make_a_root is necessary as there doesn't seem to be a way to
+    // get from Input<Buffer<int32_t>> to ExternFuncArgument otherwise.
```

Comment:
We should fix that; I'll circle back after this lands and ensure there's a clean way to do so.